.github/workflows/snapshot-automation.yml

```diff
@@ -0,0 +1,21 @@
+name: Snapshot Release Automation
+on:
+  schedule: # schedule the job to run at 12 a.m. daily
+    - cron: "0 0 * * *"
+jobs:
+  snapshot:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: develop
+      - name: Setup JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: 8
+          distribution: "adopt"
+          cache: "maven"
+      - name: Deploy Snapshot
+        timeout-minutes: 40
+        run: mvn clean deploy -DskipTests=true'
```

Comment:
Are secrets needed here?

Reply:
Thinking the same here. But I'm not sure what types of secrets I should add. This would automatically deploy to the Apache repository. The closest one I've referred to is: https://github.com/apache/dubbo/blob/3.1/Jenkinsfile

Reply:
Create a JIRA ticket, requesting them to add GitHub Secret for you. The secret will be accessible to GitHub Actions. 
Note, remember to specify the name of the repository to add to when creating tickets.

For the maven part, use variable substitution as described here: https://stackoverflow.com/questions/28071697/is-it-possible-to-pass-a-password-in-maven-deploy-in-the-command-line.

Reply:
Here is an example. https://issues.apache.org/jira/browse/INFRA-23612?filter=-2

The actual secret Key=Value is expected to be mailed to root@apache.org privately

Reply:
I've now created a JIRA ticket for it: https://issues.apache.org/jira/browse/INFRA-23708 and waiting for the response from the Infra team.